features/admin.tenants.v1/api/tenants.ts

```diff
@@ -160,3 +167,29 @@ export const getAssociatedTenants = (
             return Promise.reject(error?.response?.data);
         });
 };
+
+export const useDeploymentUnits =
+    <Data = DeploymentUnitResponse, Error = RequestErrorInterface>
+    (
+        requestOptions: SWRConfig<Data, Error>,
+        shouldFetch: boolean = false
+    ) : RequestResultInterface<Data, Error> => {
+        const requestConfig: RequestConfigInterface = {
+            headers: {
+                "Content-Type": "application/json"
+            },
+            method: HttpMethods.GET,
+            url: store.getState().config.endpoints.deploymentUnits + getDomainQueryParam()
+        };
+
+        const { data, error, isValidating, mutate } = useRequest<Data, Error>(
+            shouldFetch ? requestConfig : null, requestOptions);
+
+        return {
+            data,
+            error: error,
+            isLoading: !error && !data,
```

Comment:
We actually don't need this manual computation. SWR resolves the isLoading status for us.

```suggestion
            isLoading,
```

Reply:
There is no variable called isLoading in the context. We are setting the isLoading based on the results and return. 

The other places we use SWR also used the same approach to return the isLoading

Reply:
> There is no variable called isLoading in the context. We are setting the isLoading based on the results and return.

We can get `isLoading` from the useRequest hook.

```js
const { data, error, isLoading, isValidating, mutate } = useRequest<Data, Error>(
            shouldFetch ? requestConfig : null, requestOptions);
```

> The other places we use SWR also used the same approach to return the isLoading

Actually, the other places you refer follow an anti pattern. We need to refactor those into using the isLoading returned from the hook.

Reply:
Updated